dump/mapping/prepbufr_acft_profiles.py

```diff
@@ -0,0 +1,183 @@
+    
+        airTemperature = container.get('airTemperatureObsValue')
+        specificHumidity = container.get('specificHumidityObsValue')
+        wind = container.get('windNorthwardObsValue')
+    
+        ot_airTemperature = self._compute_typ_other(t_ot, airTemperature)
+        ot_specificHumidity = self._compute_typ_other(q_ot, specificHumidity)
+        ot_wind = self._compute_typ_uv(uv_ot, wind)
+    
+        self.log.debug(f'Change IALR to 0.0 if masked for bias correction.')
+        ialr = container.get('instantaneousAltitudeRate')
+        ialr_paths = container.get_paths('instantaneousAltitudeRate')
+        ialr2 = ma.array(ialr)
+    
+        ialr_bc = self._compute_ialr_if_masked(uv_ot, ialr2)
+    
+        self.log.debug(f'Update variables in container')
+        container.replace('instantaneousAltitudeRate', ialr_bc)
+        container.replace('airTemperatureObservationType', ot_airTemperature)
+        container.replace('specificHumidityObservationType', ot_specificHumidity)
+        container.replace('windObservationType', ot_wind)
+    
+        self.log.debug(f'Add variables to container')
+        container.add('sequenceNumber', sequenceNum, lon_paths)
+    
+        # Check
+        self.log.debug(f'container list (updated): {container.list()}')
+    
+        return container
+    
+    
+    def _compute_typ_other(self, typ, var):
+        """
+        Compute datatype if the variable is not wind.
+        Parameters:
+            typ: datatype
+            var: obsValue variable
+        Returns:
+            Masked array of the new datatype
+        """
+    
+        typ_var = copy.deepcopy(typ)
+        typ_var[(typ_var > 300) & (typ_var < 400)] -= 200
+        typ_var[(typ_var > 400) & (typ_var < 500)] -= 300
+        typ_var[(typ_var > 500) & (typ_var < 600)] -= 400
+    
+        return typ_var
+    
+    
+    def _compute_typ_uv(self, typ, var):
+        """
+        Compute datatype if the variable is wind.
+        Parameters:
+            typ: datatype
+            var: obsValue variable
+        Returns:
+            Masked array of the new datatype
+        """
+    
+        typ_var = copy.deepcopy(typ)
+        typ_var[(typ_var > 300) & (typ_var < 400)] -= 100
+        typ_var[(typ_var > 400) & (typ_var < 500)] -= 200
+        typ_var[(typ_var > 500) & (typ_var < 600)] -= 300
+    
+        return typ_var
+    
+    
+    def _compute_ialr_if_masked(self, typ, ialr):
+        """
+        Compute instantaneousAltitudeRate (IALR) if it is masked.
+        Parameters:
+            typ: datatype
+            ialr: instantaneousAltitudeRate
+        Returns:
+            Masked array of the updated instantaneousAltitudeRate
+        """
+    
+        ialr_bc = copy.deepcopy(ialr)
+        for i in range(len(ialr_bc)):
+            if ma.is_masked(ialr_bc[i]) and (typ[i] >= 330) and (typ[i] < 340):
+                ialr_bc[i] = float(0)
```

Comment:
What about 

```python
ialr_bc[(ialr_bc.mask) && (typ >= 330) && (typ < 340)] = 0.0
```

Reply:
Why modifying masked out values?

Reply:
It's part of GSI code to take certain 'typ' values and where the 'ialr' data is specifically missing, re-assign to zero. This code is required to stay because the rest need to stay missing/masked while others are changed to 0.
I change change it to your suggestion though.

Reply:
Good idea. Definitely need to use that, I'm just used to the other way. But will do in the future.